components/ide/jetbrains/gateway-plugin/src/main/kotlin/io/gitpod/jetbrains/gateway/GitpodConnectionProvider.kt

```diff
@@ -74,6 +76,18 @@ class GitpodConnectionProvider : GatewayConnectionProvider {
             parameters["workspaceId"]!!,
             parameters["backendPort"]
         )
+
+        val existingConnection = activeConnections.find {
+            it.gitpodHost == parameters["gitpodHost"] && it.workspaceId == parameters["workspaceId"] && it.backendPort == parameters["backendPort"]
+        }
+
+        if (existingConnection != null) {
```

Comment:
Java is concurrent you will need to synchronize this place, you can use concurrentmap it has methods which already synchronize check out other places for instance how we ensure that only connection to server api exists